school/school.validator.js

```
@@ -0,0 +1,73 @@
+/**
+ * Validates school input before saving to the database.
+ * Ensures required fields such as school name and address are present and properly formatted.
+ * Throws an error containing validation messages if any rule is violated.
+ *
+ * @function ValidateSchoolInput
+ * @param {Object} school_input - The input data for a school.
+ * @param {string} school_input.school_legal_name - Required. The official legal name of the school. Must be a string.
+ * @param {string} school_input.school_commercial_name - Required. The commercial or commonly used name of the school. Must be a string.
+ * @param {Array<Object>} school_input.address - Required. At least one address object must be present.
+ * Each address must include `street`, `city`, `province`, and `postal_code`.
+ *
+ * @throws {Error} Throws an error if any validation rule fails, with a message listing all validation issues.
+ */
+function ValidateSchoolInput(school_input) {
+  // *************** Variable to contain error
+  const error = [];
```

Fail fast principle violated, wasting CPU